source/extensions/filters/common/expr/custom_cel/custom_cel_vocabulary.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/protobuf/message_validator.h"
+
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/common/expr/context.h"
+
+#include "absl/strings/string_view.h"
+#include "eval/public/activation.h"
+#include "eval/public/cel_function_registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace Custom_Cel {
+
+using google::api::expr::runtime::Activation;
+using google::api::expr::runtime::CelFunctionRegistry;
+
+class CustomCelVocabulary {
+public:
+  CustomCelVocabulary() {}
+
+  // FillActivation can register variable sets using InsertValueProducer.
+  // Lazy function
+  virtual void fillActivation(Activation* activation, Protobuf::Arena& arena,
+                              const StreamInfo::StreamInfo& info,
+                              const Http::RequestHeaderMap* request_headers,
```

Comment:
Envoy style prefers references rather than pointers unless something is really optional. So, `const Http::RequestHeadMap&`. Same below in various places

Reply:
The function which calls fillActivation has these params for RequestHeaderMap, etc. as pointers.  And has the params for Arena and StreamInfo as references.  

But point taken.